rllib/core/optim/rl_optimizer.py

```diff
@@ -0,0 +1,72 @@
+import abc
+from typing import Mapping, Any
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.utils.annotations import (
+    OverrideToImplementCustomLogic,
+    OverrideToImplementCustomLogic_CallToSuperRecommended,
+    PublicAPI,
+)
+
+
+@PublicAPI(stability="beta")
+class RLOptimizer(abc.ABC):
+    """Base class for defining a loss function and optimizer for a RLModule.
+
+    Args:
+        rl_module: The RLModule that will be optimized.
+        config: The configuration for the optimizer.
+
+    Abstract Methods:
+        compute_loss: computing a loss to optimize rl_module over.
+        construct_optimizers: constructing the optimizers for rl_module.
+
+    Example:
+    .. code-block:: python
+
+        module = RLModule(...)
+        rl_optim = RLOptimizer(module, config)
+        optimizers = rl_optim.construct_optimizers()
+        sample_batch = ...
+        fwd_out = module.forward_train(sample_batch)
+        loss_dict = rl_optim.compute_loss(fwd_out)
+
+        # compute gradients of loss w.r.t. trainable variables
+        ...
+
+        for optim in optimizers:
+            optim.step()
+            optim.zero_grad()
+
+    """
+
+    @OverrideToImplementCustomLogic_CallToSuperRecommended
+    def __init__(self, module: RLModule, config: Mapping[str, Any]):
```

Comment:
Dumb question: Why not use type `Dict` here (instead of `Mapping`)?

Reply:
you can pass really any object that supports mapping criteria:
`__getitem__,` `__len__` , `__iter__`.

Also I wanted to be consistent with the RLModule documentation.

Reply:
Trying to stay consistent with the RLModule documentation